app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java

```diff
@@ -84,7 +85,13 @@ public void addCategory(Category category) {
 
     @Override
     public void onNewCategoryAdded(Intent data) {
-        //todo - Add category
+        String categoryName = data.getStringExtra("NAME");
+        String categoryDescription = data.getStringExtra("DESCRIPTION");
```

Comment:
Actually this is a good catch, Intent is an android type object so we dont want to move the whole intent into the presenter, we would parse the data out in Activity and then pass the parsed data into the presenter.